README.md

```diff
@@ -36,25 +36,62 @@ The repository contains native code project required for the Windows installer.
 
-# Building (source-build)
+# Build .NET from source (source-build)
 
-This repo also contains code to help you build the entire .NET product end-to-end from sources (often referred to as source-build), even in disconnected/offline mode. This is currently only tested on Linux.
+This repo also contains code to help you build the entire .NET product end-to-end from sources (often referred to as source-build), even in disconnected/offline mode.
+This is currently only supported on Linux.
+Please see the [dotnet/source-build](https://github.com/dotnet/source-build) repo for more information.
 
-1. `./build.sh /p:ArcadeBuildTarball=true /p:TarballDir=/path/to/place/complete/dotnet/sources`
+## Prerequisites
 
-   This fetches the complete set of source code used to build the .NET SDK. It creates a tarball of the complete .NET source code at `artifacts/packages/<Release|Debug>/Shipping/`. It also places the extracted sources at `/path/to/place/complete/dotnet/sources`. Due to a few known issues, that source directory should be outside (and not somewhere under) this repository.
+The dependencies for building .NET from source can be found [here](https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md).
 
-2. `cd /path/to/complete/dotnet/sources`
+## Building
 
-3. `./prep.sh --bootstrap`
+1. Create a .NET source tarball.
 
-    This downloads a .NET SDK and a number of .NET packages and other prebuilts needed to build .NET from source.
+   ```bash
+   ./build.sh /p:ArcadeBuildTarball=true /p:TarballDir=/path/to/place/complete/dotnet/sources
+   ```
 
-    Eventually, we want to make it possible to bootstrap .NET 6 in which case this step can be skipped.
+   This fetches the complete .NET source code and creates a tarball at `artifacts/packages/<Release|Debug>/Shipping/`.
+   The extracted source code is also placed at `/path/to/place/complete/dotnet/sources`.
+   The source directory should be outside (and not somewhere under) the installer directory.
 
-4. `./build.sh`
+2. Prep the source to build on your distro. This downloads a .NET SDK and a number of .NET packages needed to build .NET from source.
 
-    This builds the entire .NET SDK. The resulting SDK is placed at `artifacts/$ARCH/Release/dotnet-sdk-$VERSION-$ARCH.tar.gz`.
+    ```bash
+    cd /path/to/complete/dotnet/sources
+    ./prep.sh
+    ```
+    
+    On arm64, please use `./prep.sh --bootstrap` instead. This issue is being tracked [here](https://github.com/dotnet/source-build/issues/2758).
+
+3. Build the .NET SDK
+
+    ```bash
+    ./build.sh
+    ```
+
+    This builds the entire .NET SDK from source.
+    The resulting SDK is placed at `artifacts/x64/Release/dotnet-sdk-6.0.100-fedora.33-x64.tar.gz`.
+
+    Optionally add the `--online` flag to add online NuGet restore sources to the build.
```

Comment:
Should be mention the `--clean-while-building` option?  This provides the best ux for the newbies.  Perhaps we should change the default to include this as the standard behavior and an option to disable it.